### PR TITLE
gh-102500: Add PEP 688 and 698 to the 3.12 release highlights

### PR DESCRIPTION
**Pull request by hugovk**
List on What’s New In Python 3.12.

Follow format of [What’s New In Python 3.11](https://docs.python.org/3.12/whatsnew/3.11.html).

* Issue: gh-102500

----
:books: Documentation preview :books:: https://cpython-previews--104174.org.readthedocs.build/en/104174/whatsnew/3.12.html

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -68,13 +68,15 @@ Summary -- Release highlights
 
 New typing features:
 
+* :pep:`688`: Making the buffer protocol accessible in Python
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.12.rst:71]: While we're here, could you also add PEP-698?

Comment (hugovk) [reply Doc/whatsnew/3.12.rst:71]: Added!

Comment (JelleZijlstra) [reply Doc/whatsnew/3.12.rst:71]: Seems like this wasn't pushed due to the GH downtime, reminder to push the change :)

Comment (JelleZijlstra) [reply Doc/whatsnew/3.12.rst:71]: Never mind, it's there now

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**